### PR TITLE
Add -s dump-xml mode

### PR DESCRIPTION
**Pull request by pprindeville**
Fairly trivial... adding `-s dump` (in parseable text mode) will be a bit more work.

### CODE REVIEW

File: lib/src/clixon_xml_io.c [hunk 1]
```diff
@@ -290,7 +290,7 @@ xml_dump1(FILE  *f,
     if (xml_type(x) != CX_ELMNT)
         return 0;
     fprintf(stderr, "%*s %s(%s)",
-            indent*3, "",
+            indent*PRETTYPRINT_INDENT, "",
```
Comment (olofhagsand) [inline lib/src/clixon_xml_io.c:293]: nice catch!

File: lib/clixon/clixon_options.h [hunk 2]
```diff
@@ -67,7 +67,8 @@ enum startup_mode_t{
     SM_INIT,           /* Initialize running state */
     SM_RUNNING,        /* Commit running db configuration into running state */
     SM_STARTUP,        /* Commit startup configuration into running state */
-    SM_RUNNING_STARTUP /* First try running db, if it is empty try startup db */
+    SM_RUNNING_STARTUP, /* First try running db, if it is empty try startup db */
+    SM_DUMP_XML        /* Dump XML configuration */
```
Comment (olofhagsand) [inline lib/clixon/clixon_options.h:71]: + on stdout and exit

**Comment by olofhagsand** [comment]
Needs regression test update, such as amending test_startup.sh as well as user-guide update, eg https://clixon-docs.readthedocs.io/en/latest/backend.html#startup

File: lib/clixon/clixon_options.h (hunk 2 above)
Comment (pprindeville) [reply lib/clixon/clixon_options.h]: Done

**Comment by pprindeville** [comment]
> Needs regression test update, such as amending test_startup.sh as well as user-guide update, eg https://clixon-docs.readthedocs.io/en/latest/backend.html#startup

User guide updated in https://github.com/clicon/clixon-docs/pull/4.

**Pull request merged**